docs/usage/redux-logic-thunks.md

```diff
@@ -0,0 +1,443 @@
+---
+id: writing-logic-thunks
+title: Writing Logic with Thunks
+description: 'Usage > Redux Logic > Thunks: writing logic that interacts with the store'
+---
+
+:::tip What You'll Learn
+
+- What "thunks" are, and why they're used for writing Redux logic
+- How the thunk middleware works
+- Techniques for writing sync and async logic in thunks
+- Common thunk usage patterns
+
+:::
+
+## Thunk Overview
+
+### What is a "thunk"?
+
+The word "thunk" is a programming term that means ["a piece of code that does some delayed work"](https://en.wikipedia.org/wiki/Thunk). Rather than execute some logic _now_, we can write a function body or code that can be used to perform the work _later_.
+
+For Redux specifically, "thunks" are a pattern of writing logic that can interact with a Redux store, and require the [`redux-thunk` middleware](https://github.com/redujs/redux-thunk) to be added to the Redux store as part of its configuration.
+
+Thunks are [the standard approach for writing async logic in Redux apps](../style-guide/style-guide.md#use-thunks-for-async-logic), and are commonly used for data fetching. However, they can be used for a variety of tasks, and can contain both synchronous and asynchronous logic.
+
+### Writing Thunks
+
+A _thunk function_ is a function that accepts two arguments: the Redux store `dispatch` method, and the Redux store `getState` method. Thunk functions are not directly called by application code. Instead, they are passed to `store.dispatch()`:
```

Comment:
I have never heard this term used before, but I know what you mean. It's nice to see it defined!

Reply:
There's been a ridiculous amount of back and forth over "what's the thunk? is it the inner function, or the outer function?"  I finally settled on these.

Reply:
I like that and in the example you have `doThing` and `doThingThunk` which is how I've always thought of it as well